release/release_after_kokoro.sh

```diff
@@ -69,21 +70,62 @@ if [ 0 -eq $( ls -1 $SIGNED_DIR | wc -l ) ]; then
     exit 1
 fi
 
+echo "#"
+echo "# Now verifying whether jars are all signed."
+echo "#"
+for jar in $SIGNED_DIR/plugins/com.google.cloud.tools.eclipse.*.jar \
+        $SIGNED_DIR/features/com.google.cloud.tools.eclipse.*.jar; do
+    echo -n "$( basename $jar ): "
+    if ! jarsigner -strict -verify $jar; then
+        echo "unsigned artifact. Halting."
+        exit 1
+    fi
+done
+
 ###############################################################################
 echo
 echo "#"
 echo "# Verify if constants have been injected..."
 echo "#"
 ask_proceed
 
-set -x
-javap -private -classpath $SIGNED_DIR/plugins/com.google.cloud.tools.eclipse.appengine.login_0.1.0.*.jar \
-       -constants com.google.cloud.tools.eclipse.appengine.login.Constants
-javap -classpath $SIGNED_DIR/plugins/com.google.cloud.tools.eclipse.usagetracker_0.1.0.*.jar \
-       -constants com.google.cloud.tools.eclipse.usagetracker.Constants
-set +x
+LOGIN_CONSTANTS=$( javap -private -classpath \
+    $SIGNED_DIR/plugins/com.google.cloud.tools.eclipse.appengine.login_*.jar \
+    -constants com.google.cloud.tools.eclipse.appengine.login.Constants | \
+    grep OAUTH_CLIENT_ )
+ANALYTICS_CONSTANT=$( javap -classpath \
+    $SIGNED_DIR/plugins/com.google.cloud.tools.eclipse.usagetracker_*.jar \
+    -constants com.google.cloud.tools.eclipse.usagetracker.Constants | \
+    grep ANALYTICS_TRACKING_ID )
+echo -e "$LOGIN_CONSTANTS\n$ANALYTICS_CONSTANT"
+
+# Verify if the constants have been injected.
+if echo -e "$LOGIN_CONSTANTS\n$ANALYTICS_CONSTANT" | \
+        grep -q '@oauth.client\|@ga.tracking.id@'; then
+    echo "Some constant(s) have not been injected. Halting."
+    exit 1
+fi
 
-echo "Check the output above. "
+# Verify if the injected constants have the right lengths.
```

Comment:
AFAIK, the OAUTH_ strings are merely identifiers and have nothing to do with actual endpoints. They are unique, fixed identifiers generated only once for our product, CT4E.

For the Analytics tracking ID, there is no way to check if a certain ID exists or works online. It cannot be probed. Injecting the correct ID is so important, so I would like to make sure that the length is correct, to catch the situation where any of us slipped one character when copy-pasting.

I think we are okay with these brittle checks. If we had no problem exposing the constants to the public, I would have embedded the constants in the script to compare if the values are exact-match.